Solutions/InsiderRiskManagement/Analytic Rules/InsiderM365IRMAlertObserved.yaml

```diff
@@ -0,0 +1,38 @@
+id: 69660e65-0e5c-4700-8b99-5caf59786606
+name: (Private Preview) Insider Risk - Microsoft 365 Insider Risk Management Alert Observed
+description: |
+  'Microsoft 365 Insider Risk Management Triggered An Alert'
+severity: High
+requiredDataConnectors:
+  - connectorId: OfficeATP
+    dataTypes:
+      - SecurityAlert (Office 365)
+queryFrequency: 6h
+queryPeriod: 6h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - Execution
+relevantTechniques:
+  - T1204
+query: |
+  SecurityAlert
+  | where ProductName == "Microsoft 365 Insider Risk Management"
+      | summarize hint.strategy = shuffle arg_max(TimeGenerated, *), NumberOfUpdates = count() by SystemAlertId
+      | mv-expand todynamic(Entities)
+      | where Entities["Type"] =~ "account"
+      | extend Name = tostring(tolower(Entities["Name"])), NTDomain = tostring(Entities["NTDomain"]), UPNSuffix = tostring(Entities["UPNSuffix"]), AadUserId = tostring(Entities["AadUserId"]), AadTenantId = tostring(Entities["AadTenantId"]), 
+          Sid = tostring(Entities["Sid"]), IsDomainJoined = tobool(Entities["IsDomainJoined"]), Host = tostring(Entities["Host"])
+      | extend UPN = iff(Name != "" and UPNSuffix != "", strcat(Name, "@", UPNSuffix), "")
+  | extend Href_ = tostring(parse_json(ExtendedLinks)[0].Href)
+  | extend UserPrincipalName = UPN
+  | distinct AlertName, ProductName, Status, AlertLink, UserPrincipalName, Tactics, TimeGenerated
+  | sort by TimeGenerated desc
+  | limit 250
```

Comment:
Since this is a "Detection" meaning it is an Analytics and not a Hunting query, it would make more sense to make sure a customer cannot get 250 results from this query as that would result in 250 alerts.  Can you look to tighten up the query or do aggregation that would help prevent this?

Reply:
Agreed, reduced